docs/shrc.md

```diff
@@ -66,43 +140,138 @@ The [-Xshareclasses](xshareclasses.md) option is highly configurable, allowing y
+
+If you want to remove a cache but allow it to be re-created when the VM restarts, use the [`-Xshareclasses:reset`](xshareclasses.md#reset) option.
 
 
 ## Support for custom class loaders
 
-Classes are shared by the bootstrap class loader internally in the OpenJ9 VM. If you are using a custom class loader, you can use the Java Helper API to find and store classes in the shared classes cache.
+Classes are shared by the bootstrap class loader internally in the VM. The OpenJ9 implementation of `java.net.URLClassLoader` is modified to use `SharedClassURLClasspathHelper` and any class loaders that extend `java.net.URLClassLoader` inherit this behavior. If you are using a custom class loader that does not extend `java.net.URLClassLoader`, you can use the Java Helper API to find and store classes in a shared classes cache.
 
-For more information, see [Using the Java Helper API](https://www.ibm.com/support/knowledgecenter/SSYKE2_8.0.0/com.ibm.java.vm.80.doc/docs/shrc_pd_helper.html).
+If a running application uses its own class loaders, you must grant these class loaders shared class permissions before they can share classes. To grant permission, add shared class permissions to the `java.policy` file by specifying the `ClassLoader` class name. Permissions can be set for `read`, `write`, or `read,write`.
 
-## Cache utilities
+If a `SecurityManager` is being used, you must grant all custom class loaders shared class permissions by using the `SharedClassesNamedPermission` method.
+
+### The Java Helper API
+
+The Java Helper API classes can be found in the [`com.ibm.oti.shared`](api-overview.md#monitoring-and-management) package.
+
+Each class loader that wants to share classes must get a `SharedClassHelper` object from a `SharedClassHelperFactory`. The `SharedClassHelper`, when created, has a one to one relationship with the class loader. That is, it belongs to the class loader that requested it and can only store classes defined by that class loader. The `SharedClassHelper` gives the class loader a simple API for finding and storing classes in the class cache to which the VM is connected. If the class loader is garbage collected, its `SharedClassHelper` is also garbage collected.
+
+The following main functions are available from the `SharedClassHelper` API:
+
+- `findSharedClass`: Used to check whether a class is already in the cache before looking for the class on the filesystem.
+- `storeSharedClass`: Used to store a class in the cache.
+- `setSharingFilter`: A filter that can be used to decide which classes are found and stored in the cache. This filter can be applied to a particular package. To apply a filter to all non-bootstrap class loaders that share classes, specify the `-Dcom.ibm.oti.shared.SharedClassGlobalFilterClass` system property on the command line.
+
+The following APIs are available for obtaining information about a shared classes cache:
```

Comment:
This is a separate sub-section than the `SharedClassHelper` stuff. I'm thinking there should be 2 headers, "The Java Shared Classes Helper API", and "The Java Shared Classes Utility API".